include/dsn/internal/rpc_message.h

```diff
@@ -105,27 +116,27 @@ namespace dsn
         uint32_t       hdr_crc32;
         uint32_t       body_length;
         uint32_t       body_crc32;
-        uint64_t       id;      // sequence id, can be used to track source
-        uint64_t       rpc_id;  // correlation id for connecting rpc caller, request, and response tasks
+        uint64_t       id;      // sequence id
+        uint64_t       trace_id;  // used for tracking source
         char           rpc_name[DSN_MAX_TASK_CODE_NAME_LENGTH];
-        fast_rpc_name  rpc_name_fast;
+        fast_code      rpc_code; // dsn::task_code
         dsn_gpid       gpid;    // global partition id
         dsn_msg_context_t context;
         rpc_address       from_address; // always ipv4/v6 address,
                                         // generally, it is the from_node's primary address, except the
                                         // case described in message_ex::create_response()'s ATTENTION comment.
                                         // the from_address is always the orignal client's address, it will
                                         // not be changed in forwarding request.
-
         struct
         {
             uint64_t hash; // for both partition hash and thread hash for the exact location of this request
-            int32_t  timeout_ms;            
+            int64_t  timeout_ms;
```

Comment:
should be int32_t, if this is for padding, we can add int32_t padding after this field.